src/com/stanimiriliev/chaptersix/pagebean/PageBean.java

```diff
@@ -0,0 +1,143 @@
+package com.stanimiriliev.chaptersix.pagebean;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class receive list and divide it to pages
+ * @author Stanimir Iliev
+ * @since 24.07.2017
+ */
+public class PageBean <T>{
+
+    private final List<T> src;
+    public final int pageSize;
+    private int upIndex = -1;
+    private int downIndex;
+    private int pageIndex = -1;
+    private boolean hasNext = true, hasPrevious = false;
+    private List<T> list;
+
+    public PageBean(List<T> src, int pageSize){
+        if(src == null || src.size() == 0 || pageSize <= 0)
+            throw new IllegalArgumentException("This error may occur because of several things:\n" +
+                    "Source list is null\nSource list is empty\nPage size is less than or equal to zero");
+        this.src = src;
+        this.pageSize = pageSize;
+    }
+
+    /**
+     * Check if the downIndex or upIndex has reached the its limits
+     * and updates hasPrevious and hasNext variables
+     */
+    private void check(){
+        if(downIndex == 0)
+            hasPrevious = false;
+        if(upIndex == src.size() - 1)
+            hasNext = false;
+        if(upIndex < src.size() - 1 && downIndex > 0){
+            hasNext = true;
+            hasPrevious = true;
+        }
+    }
+
+    /**
+     * Reverses order of the elements of the list
+     * @param list -> The list which will be reversed
+     * @return reversed list
+     */
+    private List<T> reverseList(List<T> list){
+        List<T> temp = new ArrayList<>();
+        for(int i = list.size() - 1; i >= 0; i--)
+            temp.add(list.get(i));
+        return temp;
+    }
+
+    /**
+     * Go to the next page
+     * @return list with the next n elements of the source list (n -> page size)
+     */
+    public List<T> next(){
+        if(hasNext){
+            list = new ArrayList<>();
+            for(int i = downIndex = upIndex + 1; i < src.size() && i < upIndex + 1 + pageSize; i++)
```

Comment:
int i = downIndex = upIndex + 1 :   this is very nice.
it will be better if you define the border start and end of the page before the loop. Like that the code will look more readable.
Also if you need some other checks you can do them before the loop too.
```java
int start = currentPage * pageSize;
if(start > list.size()){
  System.out.println("There is no next page!");
  return new ArrayList<T>();
}
int end  = (currentPage +1)  * pageSize
if( end > list.size() ){
   end = list.size()
}
currentPage ++;
return list.sublist(start, end)
```
something like that...

When you have the currentPage (maybe your pageIndex) then you can easily calculate the start and the end indexes depend on list size and the page size. keeping so many states makes the task harder.